samples/headless-ssr/commerce-express/src/server.ts

```diff
@@ -0,0 +1,41 @@
+import express from 'express';
+import {renderApp} from './common/renderApp.js';
+import {renderHtml} from './common/renderHtml.js';
+import {searchEngineDefinition} from './lib/engine-definition.js';
+import {getNavigatorContext} from './lib/navigatorContext.js';
+import {middleware} from './middleware.js';
+
+const app = express();
+const port = process.env.PORT || 3000;
+app.use(middleware);
+app.use(express.static('dist'));
+
+app.get('/', async (req, res) => {
+  try {
+    searchEngineDefinition.setNavigatorContextProvider(() =>
+      getNavigatorContext(req)
+    );
+
+    const staticState = await searchEngineDefinition.fetchStaticState({
+      controllers: {
+        searchBox: {initialState: {value: req.query.q || ''}},
+        parameterManager: {
+          initialState: {parameters: {q: req.query.q || ''}},
+        },
+        productList: {initialState: {parameters: {q: req.query.q || ''}}},
+      },
+    });
```

Comment:
```suggestion
    const queryFromRequest = req.query.q ?? '';
    const staticState = await searchEngineDefinition.fetchStaticState({
      controllers: {
        searchBox: {initialState: {value: queryFromRequest}},
        parameterManager: {
          initialState: {parameters: {q: queryFromRequest }},
        },
        productList: {initialState: {parameters: {q: queryFromRequest }}},
      },
    });
```

I'm 50/50 on `||` vs `??`tho.